### ISSUE TITLE
UNKNOWN_CODE_PLEASE_REPORT: Unable to create or change a table without a primary key, when the system variable 'sql_require_primary_key' is set.

### ISSUE THREAD

**Issue by dimitrisganotis** [strapi/strapi#15329]
### Required System information

- Node.js version: 17.9.1
- NPM version: 8.11.0
- Strapi version: 4.5.5
- Database: MySQL v8 (Digital ocean managed db)
- Operating system: Ubuntu 18.04.6 LTS

### Describe the bug

I can not update the data of an admin user. 

### Steps to reproduce the behavior

1. Go to '/admin/settings/users'
2. Try to change the password of an admin user
3. See error

### Expected behavior

You should be able to update a user's password or any other data.

### Error

CREATE TABLE `orderTable_1672698304652_6112f5436b8e352a44e7c7780e703934`
    SELECT id, (
        SELECT count(*)
        FROM `admin_users_roles_links` b
        WHERE a.`role_order` >= b.`role_order` AND a.`user_id` = b.`user_id` AND a.`user_id` = 1
    ) AS src_order
    FROM `admin_users_roles_links` a 
    - UNKNOWN_CODE_PLEASE_REPORT: Unable to create or change a table without a primary key, when the system variable 'sql_require_primary_key' is set. Add a primary key to the table or unset this variable to avoid this message. Note that tables without a primary key can cause performance problems in row-based replication, so please consult your DBA before changing this setting.
Error: UNKNOWN_CODE_PLEASE_REPORT: Unable to create or change a table without a primary key, when the system variable 'sql_require_primary_key' is set. Add a primary key to the table or unset this variable to avoid this message. Note that tables without a primary key can cause performance problems in row-based replication, so please consult your DBA before changing this setting.
at Query.Sequence._packetToError (/workspace/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
at Query.ErrorPacket (/workspace/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
at Protocol._parsePacket (/workspace/node_modules/mysql/lib/protocol/Protocol.js:291:23)
at Parser._parsePacket (/workspace/node_modules/mysql/lib/protocol/Parser.js:433:10)
at Parser.write (/workspace/node_modules/mysql/lib/protocol/Parser.js:43:10)
at Protocol.write (/workspace/node_modules/mysql/lib/protocol/Protocol.js:38:16)
at TLSSocket.ondata (node:internal/streams/readable:777:22)
at TLSSocket.emit (node:events:527:28)
at TLSSocket.emit (node:domain:475:12)
at addChunk (node:internal/streams/readable:324:12)
at Protocol._enqueue (/workspace/node_modules/mysql/lib/protocol/Protocol.js:144:48)
at Connection.query (/workspace/node_modules/mysql/lib/Connection.js:198:25)
at /workspace/node_modules/knex/lib/dialects/mysql/index.js:132:18
at new Promise (<anonymous>)
at Client_MySQL._query (/workspace/node_modules/knex/lib/dialects/mysql/index.js:126:12)
at executeQuery (/workspace/node_modules/knex/lib/execution/internal/query-executioner.js:37:17)
at Client_MySQL.query (/workspace/node_modules/knex/lib/client.js:146:12)
at /workspace/node_modules/knex/lib/execution/transaction.js:363:24
at new Promise (<anonymous>)
at Client_MySQL.trxClient.query (/workspace/node_modules/knex/lib/execution/transaction.js:358:12)

**Comment by kyawthura-gg** [comment]
Downgrading to `4.5.3` fixed for me

**Comment by fabianmu** [comment]
I have the same after upgrading to 4.5.5. Running Strapi in a DigitalOcen app, node v16.19.0 as well on a MySQL v8 (Digital ocean managed db)

```
error: 
[web] [2023-01-03 18:20:43]           CREATE TABLE `orderTable_1672770043370_ea6288f21493ed0dd0c9c28764d90b04`
[web] [2023-01-03 18:20:43]             SELECT
[web] [2023-01-03 18:20:43]               id,
[web] [2023-01-03 18:20:43]               (
[web] [2023-01-03 18:20:43]                 SELECT count(*)
[web] [2023-01-03 18:20:43]                 FROM `unitgroups_components` b
[web] [2023-01-03 18:20:43]                 WHERE a.`order` >= b.`order` AND a.`entity_id` = b.`entity_id` AND a.`entity_id` = 33
[web] [2023-01-03 18:20:43]               ) AS src_order
[web] [2023-01-03 18:20:43]             FROM `unitgroups_components` a - UNKNOWN_CODE_PLEASE_REPORT: Unable to create or change a table without a primary key, when the system variable 'sql_require_primary_key' is set. Add a primary key to the table or unset this variable to avoid this message. Note that tables without a primary key can cause performance problems in row-based replication, so please consult your DBA before changing this setting.
[web] [2023-01-03 18:20:43] Error: UNKNOWN_CODE_PLEASE_REPORT: Unable to create or change a table without a primary key, when the system variable 'sql_require_primary_key' is set. Add a primary key to the table or unset this variable to avoid this message. Note that tables without a primary key can cause performance problems in row-based replication, so please consult your DBA before changing this setting.
[web] [2023-01-03 18:20:43]     at Query.Sequence._packetToError (/workspace/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
[web] [2023-01-03 18:20:43]     at Query.ErrorPacket (/workspace/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
[web] [2023-01-03 18:20:43]     at Protocol._parsePacket (/workspace/node_modules/mysql/lib/protocol/Protocol.js:291:23)
[web] [2023-01-03 18:20:43]     at Parser._parsePacket (/workspace/node_modules/mysql/lib/protocol/Parser.js:433:10)
[web] [2023-01-03 18:20:43]     at Parser.write (/workspace/node_modules/mysql/lib/protocol/Parser.js:43:10)
[web] [2023-01-03 18:20:43]     at Protocol.write (/workspace/node_modules/mysql/lib/protocol/Protocol.js:38:16)
[web] [2023-01-03 18:20:43]     at TLSSocket.ondata (node:internal/streams/readable:754:22)
[web] [2023-01-03 18:20:43]     at TLSSocket.emit (node:events:513:28)
[web] [2023-01-03 18:20:43]     at TLSSocket.emit (node:domain:489:12)
[web] [2023-01-03 18:20:43]     at addChunk (node:internal/streams/readable:315:12)
[web] [2023-01-03 18:20:43]     --------------------
[web] [2023-01-03 18:20:43]     at Protocol._enqueue (/workspace/node_modules/mysql/lib/protocol/Protocol.js:144:48)
[web] [2023-01-03 18:20:43]     at Connection.query (/workspace/node_modules/mysql/lib/Connection.js:198:25)
[web] [2023-01-03 18:20:43]     at /workspace/node_modules/knex/lib/dialects/mysql/index.js:132:18
[web] [2023-01-03 18:20:43]     at new Promise (<anonymous>)
[web] [2023-01-03 18:20:43]     at Client_MySQL._query (/workspace/node_modules/knex/lib/dialects/mysql/index.js:126:12)
[web] [2023-01-03 18:20:43]     at executeQuery (/workspace/node_modules/knex/lib/execution/internal/query-executioner.js:37:17)
[web] [2023-01-03 18:20:43]     at Client_MySQL.query (/workspace/node_modules/knex/lib/client.js:146:12)
[web] [2023-01-03 18:20:43]     at /workspace/node_modules/knex/lib/execution/transaction.js:363:24
[web] [2023-01-03 18:20:43]     at new Promise (<anonymous>)
[web] [2023-01-03 18:20:43]     at Client_MySQL.trxClient.query (/workspace/node_modules/knex/lib/execution/transaction.js:358:12)
```

Seriously, you folks have to test your releases better. We already had an issue that the webhook payload changed without any info in a release log.

**Comment by fabianmu** [comment]
Are there any news this?

**Comment by jaccospelt** [comment]
I have the same issue, available solution is to set sql_require_primary_key to false, this is possible on Digital Ocean but not recommended as it breaks the cluster option.
To enable local testing set sql_require_primary_key to true in MySQL. I am currently stuck on 4.5.3 as well.
In my case I get the error when trying to insert a new language for an existing content item.

**Comment by dimitrisganotis** [comment]
For those that use DO Managed DB you can disable sql_require_primary_key using their API:
https://docs.digitalocean.com/products/databases/mysql/#4-march-2022

**Comment by derrickmehaffy** [comment]
This is fixed in 4.6.0, marking as closed
